### DISCUSSION TITLE
第2节课作业(15班)

### DISCUSSION THREAD

**Discussion by ironartisan** [InternLM/Tutorial#92] [作业]
提交第2节课作业

**Comment by ziptoam** [comment]
https://note.youdao.com/s/T3csNdBL
1.使用 InternLM-Chat-7B 模型生成 300 字的小故事
2.熟悉 hugging face 下载功能
3.完成浦语·灵笔的图文理解及创作部署
4.完成 Lagent 工具调用 Demo 创作部署

**Reply by Mrguanglei** [reply]
飞书https://github.com/InternLM/tutorial/discussions/92

**Comment by szhowardhuang** [comment]
1） 
![image](https://github.com/InternLM/tutorial/assets/1407300/c2e6456c-9e08-4247-8ba2-5d0ee88d1401)

2）由于开发机去不了huggingface，所以在本地win10实践一下。

![image](https://github.com/InternLM/tutorial/assets/1407300/17b50ada-9b8c-4ab0-92b1-869e134691f3)

![image](https://github.com/InternLM/tutorial/assets/1407300/5cdc88da-0915-4f3d-ae22-2438c588eb20)

3）
![image](https://github.com/InternLM/tutorial/assets/1407300/3fc6d1df-230d-4dbd-9c4c-40e21d70a43a)

![image](https://github.com/InternLM/tutorial/assets/1407300/fb36562a-b4d9-41f0-a6a8-a618cba19acd)

4) finish action错了
![image](https://github.com/InternLM/tutorial/assets/1407300/0ef54839-3708-4c71-9f55-99452caa247f)

**Comment by woodx9** [comment]
作业截图请看运行浦语大模型实操截图部分和huggingface使用部分 [飞书截图](https://gxloh9mfvxh.feishu.cn/wiki/O21zwwNL5iq7KrkEFxwcNGDAnVf?from=from_copylink)

**Comment by Mrguanglei** [comment]
飞书：https://gzr5xfkph4.feishu.cn/docx/ExRGdc6UGo37n4x5743cYbXgnqd?from=from_copylink

**Comment by Gaia-yue** [comment]
https://github.com/Gaia-yue/---/blob/main/%E4%B8%80%E6%9C%88/2_Demo.md

**Comment by dqyydq** [comment]
https://twwziubpd8l.feishu.cn/wiki/HnIvwdqwAiv7EgkG80NcD73znUh?from=from_copylink

**Comment by bblgbr** [comment]
第二次课作业，github的pdf链接：
https://github.com/bblgbr/play_llm/blob/main/InterLM_Class/02%20%E4%BD%9C%E4%B8%9A.pdf

**Comment by Taixi-Feng** [comment]
1）使用 InternLM-Chat-7B 模型生成 300 字的小故事（需截图）。
![微信图片_20240107172746](https://github.com/InternLM/tutorial/assets/104333661/348f403d-dd4e-4049-92f5-c16de99ac321)
2）熟悉 hugging face 下载功能，使用 huggingface_hub python 包，下载 InternLM-20B 的 config.json 文件到本地（需截图下载过程）。
![image](https://github.com/InternLM/tutorial/assets/104333661/34173683-ab78-4c79-b717-8234231eca52)
![989eaf749761526f860de07a05b6a56](https://github.com/InternLM/tutorial/assets/104333661/4bde5623-38ff-4cb2-ad1c-2eab294e424e)


3）完成浦语·灵笔的图文理解及创作部署（需截图）
![微信图片_20240107213705](https://github.com/InternLM/tutorial/assets/104333661/b2bbf486-f047-43bd-ba2e-647448e3fbf7)
4）完成 Lagent 工具调用 Demo 创作部署（需截图）
![微信图片_20240107215134](https://github.com/InternLM/tutorial/assets/104333661/c503cbf9-c552-4f9d-970f-c31414a85f45)

**Comment by Guan-Y** [comment]
<img width="1877" alt="lagent" src="https://github.com/InternLM/tutorial/assets/61380453/25e36b75-25ee-4f48-82ae-9886f66a7dcf">
<img width="1916" alt="webdemo" src="https://github.com/InternLM/tutorial/assets/61380453/c5a64779-74e1-4a7d-a49a-0a600763a320">
<img width="1796" alt="xcomposer多模态对话" src="https://github.com/InternLM/tutorial/assets/61380453/a2fbc2f4-e663-43ad-964a-44664b7723ef">
![科研助手文章](https://github.com/InternLM/tutorial/assets/61380453/ce5f9f6a-1506-4587-a581-06a9e91954ce)

**Comment by scchy** [comment]
第二次课作业链接：（基础+进阶）
https://github.com/scchy/My_Learn/blob/master/LLM_prime/2_InternLM_demo_HW.md

**Comment by IyGnahz** [comment]
笔记&作业：https://g6xuom3zbh.feishu.cn/docx/VwKGd08m0oxbzvxPJRScMtPhnwb

**Comment by sokolo05** [comment]
第二节课笔记：https://github.com/sokolo05/Scholar_PuYu/blob/main/02.%E8%AF%BE%E7%A8%8B%E7%AC%94%E8%AE%B0/%E7%AC%AC%E4%BA%8C%E8%8A%82_%E7%AC%94%E8%AE%B0.md

**Comment by yabgdsajghsj** [comment]
[第二课作业.docx](https://github.com/InternLM/tutorial/files/13857393/default.docx)

**Comment by lizhunkg** [comment]
第二课作业，https://note.youdao.com/s/cEgP6NTD

**Comment by lta155** [comment]
https://note.youdao.com/s/3LWEY5kn

**Comment by JakChen797** [comment]
https://juejin.cn/post/7321271017430597684

**Comment by pyajagod** [comment]
第二次课作业链接
https://juejin.cn/post/7321550265609535539

**Comment by 6forwater29** [comment]
https://blog.csdn.net/weixin_44468844/article/details/135455228?csdn_share_tail=%7B%22type%22%3A%22blog%22%2C%22rType%22%3A%22article%22%2C%22rId%22%3A%22135455228%22%2C%22source%22%3A%22weixin_44468844%22%7D

**Comment by 520Enterprise** [comment]
笔记&作业：https://www.cnblogs.com/520Enterprise/p/17945399

**Comment by Visytudz** [comment]
[2 InternLM Demo HW.pdf](https://github.com/InternLM/tutorial/files/13860203/2.InternLM.Demo.HW.pdf)

**Comment by huiruije** [comment]
笔记+作业
https://blog.csdn.net/m0_59356572/article/details/135466423?spm=1001.2014.3001.5501

**Comment by huiruije** [comment]
笔记+作业
https://blog.csdn.net/m0_59356572/article/details/135466423?spm=1001.2014.3001.5501

**Comment by huiruije** [comment]
笔记+作业
https://blog.csdn.net/m0_59356572/article/details/135466423?spm=1001.2014.3001.5501

**Comment by huiruije** [comment]
笔记+作业
https://blog.csdn.net/m0_59356572/article/details/135466423?spm=1001.2014.3001.5501

**Comment by huiruije** [comment]
笔记+作业
https://blog.csdn.net/m0_59356572/article/details/135466423?spm=1001.2014.3001.5501

**Comment by huiruije** [comment]
笔记+作业
https://blog.csdn.net/m0_59356572/article/details/135466423?spm=1001.2014.3001.5501

**Comment by HarveyLiii** [comment]
1. InternLM-Chat-7B 智能对话
    - 终端运行
![截屏2024-01-08 22 25 40](https://github.com/InternLM/tutorial/assets/127242206/97e82837-9bde-43e5-832c-fba090e9b0da)
    - 网页运行
![截屏2024-01-08 22 29 58](https://github.com/InternLM/tutorial/assets/127242206/080024aa-a559-48b8-be1e-f5f293b7d600)
1. Lagent工具调用解简单数学题
![截屏2024-01-08 22 43 44](https://github.com/InternLM/tutorial/assets/127242206/df974c54-140f-488e-a7ac-9852b58821d8)
1. 浦语·灵笔多模态图文创作和理解
    - 文章创作
![截屏2024-01-08 23 13 31](https://github.com/InternLM/tutorial/assets/127242206/4f9d9d9a-415b-4c3b-9596-05d80a644f15)
    - 图片理解
![截屏2024-01-08 23 17 03](https://github.com/InternLM/tutorial/assets/127242206/46653291-f0cd-46ca-85cc-feecc0a00aad)

**Comment by cd271** [comment]
编写故事：
![1921704815100_ pic](https://github.com/InternLM/tutorial/assets/116601665/ba3070b5-3b91-4fdf-8de7-66a67bc2bec5)
下载模型报错：

![1931704815840_ pic](https://github.com/InternLM/tutorial/assets/116601665/031355f2-122b-4ef8-aa03-0207e034defe)

**Comment by 6forwater29** [comment]
笔记+作业
https://blog.csdn.net/weixin_44468844/article/details/135480134?csdn_share_tail=%7B%22type%22%3A%22blog%22%2C%22rType%22%3A%22article%22%2C%22rId%22%3A%22135480134%22%2C%22source%22%3A%22weixin_44468844%22%7D

**Comment by seienn** [comment]
https://gvxxiiwsufy.feishu.cn/wiki/EVCOwkPBFiXK5MkECrbcaadSn7d?from=from_copylink

**Comment by NANDSS1** [comment]
https://www.yuque.com/nandss/gyyg9p/ubzdofcixocvnz2g?singleDoc# 《lesson 2 书生·浦语大模型 Demo作业》

**Comment by g29times** [comment]
https://novagen.notion.site/6c6ee58af2ad49fb902003ab1392b8f4
